pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerBase.java

```diff
@@ -616,6 +642,25 @@ public CompletableFuture<Void> acknowledgeCumulativeAsync(MessageId messageId, T
         return doAcknowledgeWithTxn(messageId, AckType.Cumulative, Collections.emptyMap(), txnImpl);
     }
 
+    protected CompletableFuture<Void> acknowledgeCumulativeAsync(Map<String, MessageId> topicToMessageIdMap) throws PulsarClientException {
+        return acknowledgeCumulativeAsync(topicToMessageIdMap, null);
+    }
+
+    protected CompletableFuture<Void> acknowledgeCumulativeAsync(Map<String, MessageId> topicToMessageIdMap, Transaction txn) throws PulsarClientException {
+        validateTopicToMessageIdMap(topicToMessageIdMap);
+        if (!isCumulativeAcknowledgementAllowed(conf.getSubscriptionType())) {
+            return FutureUtil.failedFuture(new PulsarClientException.InvalidConfigurationException(
+                    "Cannot use cumulative acks on a non-exclusive/non-failover subscription"));
+        }
+
+        TransactionImpl txnImpl = null;
```

Comment:
It looks like we are not covering the TC branch with the new tests

Reply:
I am wary of the transaction usecase in cummulative ack over multiple partitions.
1. I have removed acknowledgeCumulativeAsync(Map<String, MessageId> topicToMessageIdMap, Transaction txn) function i added, as it is not primary aim of this PR. please guide me if it is valid usecase. I will add support in Consumer.java Interface and write testcase
2. I have added testcase for negative input case